op-node/rollup/driver/state.go

```diff
@@ -381,18 +373,41 @@ func (s *Driver) ResetDerivationPipeline(ctx context.Context) error {
 	}
 }
 
+func (s *Driver) syncStatus() *eth.SyncStatus {
+	return &eth.SyncStatus{
+		CurrentL1:          s.derivation.Origin(),
+		CurrentL1Finalized: s.derivation.FinalizedL1(),
+		HeadL1:             s.l1State.L1Head(),
+		SafeL1:             s.l1State.L1Safe(),
+		FinalizedL1:        s.l1State.L1Finalized(),
+		UnsafeL2:           s.derivation.UnsafeL2Head(),
+		SafeL2:             s.derivation.SafeL2Head(),
+		FinalizedL2:        s.derivation.Finalized(),
+	}
+}
+
 func (s *Driver) SyncStatus(ctx context.Context) (*eth.SyncStatus, error) {
-	respCh := make(chan eth.SyncStatus, 1)
+	wait := make(chan struct{})
 	select {
+	case s.stateReq <- wait:
+		resp := s.syncStatus()
+		<-wait
+		return resp, nil
 	case <-ctx.Done():
 		return nil, ctx.Err()
-	case s.syncStatusReq <- respCh:
-		select {
-		case <-ctx.Done():
-			return nil, ctx.Err()
-		case resp := <-respCh:
-			return &resp, nil
-		}
+	}
+}
+
+func (s *Driver) BlockRefWithStatus(ctx context.Context, num uint64) (eth.L2BlockRef, *eth.SyncStatus, error) {
+	wait := make(chan struct{})
+	select {
+	case s.stateReq <- wait:
+		resp := s.syncStatus()
+		ref, err := s.l2.L2BlockRefByNumber(ctx, num)
```

Comment:
Why do we need to run block by number through the event loop?

Reply:
We want block-by-number to be consistent with the returned status - reorgs may affect by-number requests.